python/hail/expr/expressions/typed_expressions.py

```diff
@@ -1352,6 +1352,10 @@ def drop(self, *fields):
         return construct_expr(StructOp('drop', self._ast, *to_drop), result_type,
                               self._indices, self._aggregations, self._joins)
 
+    @typecheck_method(other=expr_struct())
+    def concat(self, other):
```

Comment:
I like this, but it needs a short doc and you should decide what to do if there's a duplicate field name in both sides. Currently it'll error with a somewhat nasty message about passing the same keyword arg twice.

Reply:
i was thinking about this and I think `annotate` does most of what we want this to do?

Reply:
I've taken it out in favor of an explicit call to `hl.struct(**k, **v)` in the places where it's used. (They're all internal usages where an error would have been a bug). I think `hl.annotate(s1, s2)` does what this would have done for non-overlapping structs, with better-defined behavior for overlapping structs. I could also keep this in and do field-checking explicitly here to throw a better error, if that would still be useful.